src/DataTab.ts

```diff
@@ -0,0 +1,69 @@
+import { CheckBoxParams, ComboBoxParams, ControlCreator, DropBoxParams, IListModifiable, IObjectInspectorControl, IObjectInspectorSpecification, ListModifierParams } from '@displayr/ngviz';
+
+export type DataTabConfig = {
+    inputSelector: DropBoxParams;
+    switchRowColumn?: CheckBoxParams;
+    showNetSum?: ComboBoxParams;
+    labelModifier?: ListModifierParams;
+};
+
+export type DataTabControls = {
+    inputSelector: IObjectInspectorControl<string[]>;
+    switchRowColumn?: IObjectInspectorControl<boolean>;
+    showNetSum?: IObjectInspectorControl<string>;
+    labelModifier?: IObjectInspectorControl<IListModifiable[]>;
+}
+
+/** Proof of concept for a generalised method that handles the construction of the Data Tab, intended to be used
+ * by every NGViz. Will eventually be moved to ngviz-common */
+export function constructDataTab(
+    config: DataTabConfig,
+    settings: IObjectInspectorSpecification,
+): DataTabControls {
+    // This page will eventually be called 'Data' instead of 'Inputs'
+    const page = 'Inputs';
+
+    const primary_data = settings.dropBox({
+        name: 'primaryData',
+        label: 'Data',
+        page,
+        group: 'Data Source',
+        ...config.inputSelector,
+    });
+
+    let controls: DataTabControls = { inputSelector: primary_data };
+
+    if (config?.switchRowColumn) {
+        const switch_rows = settings.checkBox({
+            name: 'switchRowsColumns',
+            label: 'Switch rows/columns',
+            page,
+            group: 'Data Source',
+            ...config.switchRowColumn,
+        });
+        controls = { ...controls, switchRowColumn: switch_rows };
```

Comment:
Is there a reason we've chosen to reassign the variable instead of mutate the object? They both mutate something, I wonder if the latter is just easier on the eyes:

```suggestion
        controls.switchRowColumn = switch_rows;
```

Reply:
Happy to update this